### PR TITLE
RUMM-247 RUM Resource Interceptor

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Create an Interceptor to log RUM Resources loading

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        val url = request.url().toString()
+        val method = request.method()
+
+        GlobalRum.get().startResource(request, url, mapOf("http.method" to method))
+
+        try {
+            val response = chain.proceed(request)
+            GlobalRum.get().stopResource(
+                request,
+                response.header(HEADER_CT),
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt]: For now I didn't see "mimeType" being used in the "stopResource" method. Is this for later ?

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * This interceptor will log the request as a RUM resource fill the request information
+ * (url, method, status code, optional error).
+ *
+ * If you use multiple Interceptors, make sure that this one is called first.
+ *
+ * To use:
+ * ```
+ *   OkHttpClient client = new OkHttpClient.Builder()
+ *       .addInterceptor(new TracingInterceptor())
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt]: Indeed 🤦‍♂

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt]: Yes, that's teh next PR ;)

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumInterceptor.kt]: Done

**Pull request merged**